meteor/lib/api/rundownLayouts.ts

```diff
@@ -29,7 +35,161 @@ export enum RundownLayoutsAPIMethods {
 	'createRundownLayout' = 'rundownLayout.createRundownLayout',
 }
 
+export interface LayoutDescriptor<T extends RundownLayoutBase> {
+	supportedElements: RundownLayoutElementType[]
+	filtersTitle?: string // e.g. tabs/panels
+	supportsFilters?: boolean
+}
+
+export interface CustomizableRegionSettingsManifest {
+	_id: string
+	title: string
+	layouts: Array<CustomizableRegionLayout>
+}
+
+export interface CustomizableRegionLayout {
+	_id: string
+	type: RundownLayoutType
+	filtersTitle?: string
+	supportsFilters?: boolean
+	supportedElements: RundownLayoutElementType[]
+}
+
+class RundownLayoutsRegistry {
+	private shelfLayouts: Map<RundownLayoutType, LayoutDescriptor<RundownLayoutShelfBase>> = new Map()
+	private rundownViewLayouts: Map<RundownLayoutType, LayoutDescriptor<RundownViewLayout>> = new Map()
+	private miniShelfLayouts: Map<RundownLayoutType, LayoutDescriptor<RundownLayoutShelfBase>> = new Map()
+	private rundownHeaderLayouts: Map<RundownLayoutType, LayoutDescriptor<RundownLayoutRundownHeader>> = new Map()
+
+	public RegisterShelfLayout(id: RundownLayoutType, description: LayoutDescriptor<RundownLayoutShelfBase>) {
```

Comment:
All of these method names should be camelCase.

Reply:
Addressed with https://github.com/olzzon/tv-automation-server-core/commit/e6d51bd2db4ef5d9382970c33eb5219308ee382b